firebase-common/src/main/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatInfo.java

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.heartbeatinfo;
+
+import android.content.Context;
+import androidx.annotation.RestrictTo;
+import androidx.annotation.VisibleForTesting;
+import com.google.firebase.components.Component;
+import com.google.firebase.components.Dependency;
+import com.google.firebase.internal.HeartBeatInfoStorage;
+import org.jetbrains.annotations.NotNull;
+
+/** Provides information as whether to send heart beat or not. */
+public class DefaultHeartBeatInfo implements HeartBeatInfo {
+
+  private HeartBeatInfoStorage storage;
+
+  private DefaultHeartBeatInfo(Context context) {
+    HeartBeatInfoStorage.initialize(context);
+    storage = HeartBeatInfoStorage.getInstance();
+  }
+
+  @VisibleForTesting
+  @RestrictTo(RestrictTo.Scope.TESTS)
+  DefaultHeartBeatInfo(HeartBeatInfoStorage testStorage) {
+    storage = testStorage;
+  }
+
+  @Override
+  public @NotNull HeartBeat getHeartBeatCode(@NotNull String heartBeatTag) {
+    long presentTime = System.currentTimeMillis();
+    boolean shouldSendSdkHB = storage.shouldSendSdkHeartBeat(heartBeatTag, presentTime);
+    boolean shouldSendGlobalHB = storage.shouldSendGlobalHeartBeat(presentTime);
+    if (!shouldSendSdkHB && !shouldSendGlobalHB) {
```

Comment:
it may be simpler if you did:
```
if (shouldSendSdk && shouldSendGlobal) {
  return HeartBeat.COMBINED;
} else if (shouldSendGlobal) {
  return HeartBeat.GLOBAL;
} else if (shouldSendSdk) {
  return HeartBeat.SDK;
}
return HeartBeat.NONE;
```